### PR TITLE
Stop using the obsolete snapcraft-desktop-helpers

### PR DESCRIPTION
**Pull request by AlanGriffiths**

### CODE REVIEW

File: snap/hooks/configure [hunk 1]
```diff
@@ -1,30 +1,27 @@
...
+# Remove legacy console-provider option
+snapctl set console-provider=null
 
 mir_demo_server_config="${XDG_CONFIG_HOME}/mir/mir_demo_server.config"
 
 # Ensure we have a config file with the fixed options
 if [ ! -e "${mir_demo_server_config}" ]; then
-mkdir -p $(dirname "${mir_demo_server_config}")
-cat <<EOT > "${mir_demo_server_config}"
+  mkdir -p "$(dirname "${mir_demo_server_config}")"
```
Comment (Saviq) [inline snap/hooks/configure]: Kinda surprised the quotes here work ¯\_(ツ)_/¯

**Review by Saviq — COMMENTED** [review]
Wanna be consistent about `{}` usage in variables?

As discussed, we probably should just get rid of the `configure` hook here.

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -104,14 +91,32 @@ slots:
...
+  /usr/share/glib-2.0/schemas:
+    bind: $SNAP/usr/share/glib-2.0/schemas
+  /usr/lib/$SNAPCRAFT_ARCH_TRIPLET/gdk-pixbuf-2.0:
+    bind: $SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/gdk-pixbuf-2.0
+  /usr/share/glmark2:
+    bind: $SNAP/usr/share/glmark2
+  /usr/share/libdrm:  # Needed by mesa-core20 on AMD GPUs
+    bind: $SNAP/graphics/libdrm
+  /usr/share/drirc.d:  # Used by mesa-core20 for app specific workarounds
+    bind: $SNAP/graphics/drirc.d
```
Comment (Saviq) [inline snap/snapcraft.yaml]: None of these can be made into environment variables?

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -177,27 +182,32 @@ parts:
...
-      sed --in-place s'/^  needs_update=true/:/' $SNAPCRAFT_PART_INSTALL/bin/desktop-launch
-
-  gtk3-apps:
-    plugin: nil
+      update-mime-database $SNAPCRAFT_PART_INSTALL/usr/share/mime
+      # Compile the gsettings schemas
+      /usr/lib/${SNAPCRAFT_ARCH_TRIPLET}/glib-2.0/glib-compile-schemas "$SNAPCRAFT_PART_INSTALL/usr/share/glib-2.0/schemas"
+      # Index the pixbuf loaders
+      GDK_PIXBUF_MODULEDIR=$SNAPCRAFT_PART_INSTALL/usr/lib/${SNAPCRAFT_ARCH_TRIPLET}/gdk-pixbuf-2.0/2.10.0/loaders
+      "/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/gdk-pixbuf-2.0/gdk-pixbuf-query-loaders" > "$SNAPCRAFT_PART_INSTALL/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/gdk-pixbuf-2.0/2.10.0/loaders.cache"
```
Comment (wmww) [inline snap/snapcraft.yaml]: I don't *think* `gdk-pixbuf-query-loaders` will see the `GDK_PIXBUF_MODULEDIR` we set on the previous line

Comment (AlanGriffiths) [reply snap/snapcraft.yaml]: Fixed

File: snap/snapcraft.yaml (hunk 2 above)
Comment (AlanGriffiths) [reply snap/snapcraft.yaml]: Fixed

**Review by Saviq — APPROVED** [review]
Yup.

bors merge

**Pull request merged**